tiiuae/ghaf-givc
# Replace wifi nmcli commands with dbus

## Description of changes

Replace the nmcli commands with dbus commands
Add disconnection function

## Testing
AP List:
- Use grpcurl to query of wifi AP list: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ListNetwork`
- Observe the network list with SSID, signal strength and security lists. 

Connect:
- Replace ssid and password in the command and run grpcurl to connect to AP: `grpcurl '{"SSID":"ssid","Password":"password"}' -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ConnectNetwork`.
- Observe the network connection on the Taskbar. 

Disconnect
- Replace ssid and password in the command and run grpcurl to connect to AP: `grpcurl  -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.DisconnectNetwork`.
- Observe the no connection message on the Taskbar. 

Turn Off:
- Use grpcurl to turn off the wifi: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.TurnOff`.
- Use grpcurl to query of wifi AP list: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ListNetwork`
- Observe the empty network list. 

Turn On:
- Repeat the connection process, Check the connection and `No network with SSID found` error message.
- Use grpcurl to turn on the wifi back: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.TurnOn`.
- Observe the successful network connection again. 
- Use grpcurl to query of wifi AP list: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ListNetwork`
- Observe the non-empty network list.
